modules/renter/hostdb/hostdb.go

```diff
@@ -39,6 +41,119 @@ var (
 	errHostNotFoundInTree = errors.New("host not found in hosttree")
 )
 
+// blockedDomains manages a list of blocked domains
+type blockedDomains struct {
+	domains  map[string]struct{}
+	resolver net.Resolver
+	mu       sync.Mutex
+}
+
+// newBlockedDomains initializes a new blockedDomains
+func newBlockedDomains(domains []string) *blockedDomains {
+	domainMap := make(map[string]struct{})
+	for _, domain := range domains {
+		domainMap[domain] = struct{}{}
+	}
+	return &blockedDomains{
+		domains: domainMap,
+	}
+}
+
+// managedAddDomains adds domains to the map of blocked domains
+func (bd *blockedDomains) managedAddDomains(domains []string) {
+	bd.mu.Lock()
+	defer bd.mu.Unlock()
+	for _, domain := range domains {
+		bd.domains[domain] = struct{}{}
+
+		addrs, err := bd.resolver.LookupHost(context.Background(), domain)
```

Comment:
could we use `net.LookupHost` here instead? Seems like the benefit of using a distinct `net.Resolver` is that you can pass a context, but we don't make use of that here.